### DISCUSSION TITLE
Inspecting previous run

### DISCUSSION THREAD

**Discussion by brianthomas-washburn** [alphaparrot/ExoPlaSim#35] [Q&A] [answered]
Apologies in advance if this is just a "I don't know python well" question.

I'm trying to use "inspect" to look at results from a run but I'm not sure how to bring in the model that was previously defined and run.

So I know its model.inspect("ts") but how does one tell the system how to access that model?

Thanks, and apologies again if this is a dumb question!

**Comment by alphaparrot** [comment, marked as answer]
Hey please don't put ChatGPT garbage in here. This answer is useless and
insulting to the intelligence of every literate person. Kindly never do
that again, ever, on any repository.

Brian, if you're asking how to get back to having that 'model' variable in
a future session, you can use the model.save function to store the model
instance in a numpy file, and then restore it with
`model=numpy.load("path/to/model.npy").item()`. The inspect function is
just a thin wrapper over some generalized access functions and utility
functions; the output files can also be accessed normally without ExoPlaSim
using h5py for HDF5 formats, NetCDF4 for .nc files, numpy for npz files,
etc.

Accessing output files with the 'inspect' utility *without* having an
instantiated model loaded into memory is a bit of a gap in the current
functionality; I hope to rectify that in 4.0.

**Reply by brianthomas-washburn** [reply]
Ok, makes sense.  Sounds like what I had in mind can't be done unless I've first used the model.save function.  I'll add that to my workflow or just use another tool to access the output.

Thanks for your reply, much appreciated!
